pkg/vetter/util/mtlspolicy/authPolicy_test.go

```diff
@@ -160,56 +193,475 @@ var (
+			&istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Mtls{
+					Mtls: &istioauthv1alpha1.MutualTls{
+						Mode: istioauthv1alpha1.MutualTls_PERMISSIVE,
+					},
+				},
+			},
+			&istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Mtls{
+					Mtls: &istioauthv1alpha1.MutualTls{},
+				},
+			},
+			&istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Jwt{},
+			},
+		}
+		peersEnabledPlusJWT := []*istioauthv1alpha1.PeerAuthenticationMethod{
+			&istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Mtls{
+					Mtls: &istioauthv1alpha1.MutualTls{},
+				},
+			},
+			&istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Mtls{},
+			},
+			&istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Jwt{},
+			},
+		}
+		peersDisabled := []*istioauthv1alpha1.PeerAuthenticationMethod{
+			&istioauthv1alpha1.PeerAuthenticationMethod{},
+			&istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Jwt{},
+			},
+		}
+		It("returns MIXED when len() == 1 && Mode is set to permissive", func() {
+			mtlsState := getModeFromPeers(peersPermissive)
+			Expect(mtlsState).To(Equal(MTLSSetting_MIXED))
+		})
+		It("returns ENABLED when len() == 1 && the Mode is STRICT", func() {
+			mtlsState := getModeFromPeers(peersStrict)
+			Expect(mtlsState).To(Equal(MTLSSetting_ENABLED))
+		})
+		It("returns MIXED when PERMISSIVE is set and there are multiple options enabling mtls", func() {
+
+			mtlsState := getModeFromPeers(peersMixed)
+			Expect(mtlsState).To(Equal(MTLSSetting_MIXED))
+		})
+		It("returns ENABLED when there are multiple options enabling auth", func() {
+			mtlsState := getModeFromPeers(peersEnabledPlusJWT)
+			Expect(mtlsState).To(Equal(MTLSSetting_ENABLED))
+		})
+		It("returns DISABLED when there are no mtls auth methods present", func() {
+			mtlsState := getModeFromPeers(peersDisabled)
+			Expect(mtlsState).To(Equal(MTLSSetting_DISABLED))
+		})
 	})
-	It("should evaluate empty-mtls-peer as true", func() {
-		Expect(AuthPolicyIsMtls(apFooPortsBarOn)).To(BeTrue())
+})
+
+var _ = Describe("paramIsMTls()", func() {
+	Context("paramIsMTls()", func() {
+		It("determines whether mtls is enabled for a Peer", func() {
+
+			peer := istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Mtls{},
+			}
+			peer2 := istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Jwt{},
+			}
+			peer3 := istioauthv1alpha1.PeerAuthenticationMethod{
+				Params: &istioauthv1alpha1.PeerAuthenticationMethod_Mtls{
+					Mtls: &istioauthv1alpha1.MutualTls{},
+				},
+			}
+
+			ok := paramIsMTls(&peer)
```

Comment:
No harm in writing these as "Expect(paramIsMtls(&peer)).To(BeTrue())" since they are so short IMO, just as an aside.

Reply:
done! :)